### ISSUE TITLE
git-checkout doesn't refer to which repository is missing a reference

### ISSUE THREAD

**Issue by jeapostrophe** [racket/racket#3329]
So it is hard to know what to fix when something fails during `make`

Here's a patch:

```
diff --git a/racket/collects/net/git-checkout.rkt b/racket/collects/net/git-checkout.rkt
index 50f4f30e06..d049c0476d 100644
--- a/racket/collects/net/git-checkout.rkt
+++ b/racket/collects/net/git-checkout.rkt
@@ -91,7 +91,7 @@
           ;; Find the commits needed for `ref`:
           (define-values (ref-commit    ; #f or an ID string
                           want-commits) ; list of ID string
-            (select-commits ref refs status try-only-master?))
+            (select-commits ref refs status try-only-master? repo))
 
           (unless dest-dir
             (write-pkt o) ; clean termination
@@ -356,7 +356,7 @@
 ;;  initial response. If we can, the list of requested IDs will be
 ;;  just that one. Otherwise, we'll have to return a list of all
 ;;  IDs, and then we'll look for the reference later.
-(define (select-commits ref refs status try-only-master?)
+(define (select-commits ref refs status try-only-master? repo)
   (define ref-looks-like-id? (regexp-match? #rx"^[0-9a-f]+$" ref))
 
   (define ref-rx (byte-regexp (bytes-append
@@ -385,14 +385,14 @@
         [try-only-master?
          (status "Requested reference looks like commit id; try within master")
          (define-values (master-ref-commit want-commits)
-           (select-commits "master" refs status #f))
+           (select-commits "master" refs status #f repo))
          want-commits]
         [else
          (status "Requested reference looks like commit id; getting all commits")
          (for/list ([ref (in-list refs)])
            (cadr ref))])]
      [else
-      (raise-git-error 'git "could not find requested reference\n  reference: ~a" ref)]))
+      (raise-git-error 'git "could not find requested reference\n  reference: ~a\n  repo: ~v" ref repo)]))
   
   (values ref-commit want-commits))
```

I can commit, @mflatt , if cool with you.

**Comment by mflatt** [comment]
I think probably `~a` is better for formatting the repo, but good otherwise.

**Comment by mflatt** [comment]
Closed by 09ff1344e8637ba857de59d93640366c863a5af5
